### PR TITLE
Encoded slashes are expanded by the proxy

### PR DESCRIPTION
**Pull request by jehiah**
To reproduce, hit this URL on the proxy: `http://proxy/a%2Fb/`

It is passed to the backend as `http://backend/a/b/`

The expected result is for the backend to receive `http://backend/a%2Fb/`

This is a problem for Jenkins and is described here: https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+says+my+reverse+proxy+setup+is+broken

### CODE REVIEW

**Comment by jehiah** [comment]
@ploxiln mind taking a look at this?

@adrian-gomez I saw you also had a similar patch for this in 3af83e0eef8a545f009d20ade7603ff77fc54e58 so If you have any comments I'd appreciate it.

**Comment by adrian-gomez** [comment]
Hi, yeah i had the issue but with gerrit. The fix as someone mentioned is to use a custom proxy instead of NewSingleHostReverseProx for now at least. The real issue is with NewSingleHostReverseProxy that is not doing a great job as a proxy.

I have fixed this in the commit you mentioned i can do a PR ir you want

**Comment by jehiah** [comment]
@adrian-gomez Yeah your commit was helpful. I've updated this PR w/ a patch and I was hoping you could help verify that a build from this branch works as expected for you.

**Comment by adrian-gomez** [comment]
Yeah np, i'll test and comment back before the end of the day.

File: oauthproxy_test.go [hunk 1]
```diff
@@ -35,3 +35,30 @@ func TestNewReverseProxy(t *testing.T) {
...
+	f, _ := url.Parse(frontend.URL)
+	encodedPath := "/a%2Fb/"
+	getReq := &http.Request{URL: &url.URL{Scheme: "http", Host: f.Host, Opaque: encodedPath}}
+	_, err := http.DefaultClient.Do(getReq)
+	if err != nil {
+		t.Fatalf("err %s", err)
+	}
+	expected := backend.URL + encodedPath
+	if seen != expected {
+		t.Errorf("got bad requst %q expected %q", seen, expected)
```
Comment (ploxiln) [inline oauthproxy_test.go]: s/requst/request/

**Comment by ploxiln** [comment]
I'm actually not using google_auth_proxy at the present moment, so take this with a grain of salt, but this LGTM.

**Pull request merged**